src/Bedrock.Framework.Experimental/Transports/WebSockets/WebSocketFrameWriter.cs

```diff
@@ -78,12 +78,23 @@ public unsafe void WriteMessage(WebSocketWriteFrame message, IBufferWriter<byte>
             if (message.Header.Masked)
             {
                 headerSpan[1] |= 0b1000_0000;
-                BinaryPrimitives.WriteInt32BigEndian(headerSpan, message.Header.MaskingKey);
+
+                //An int is used for storage, but the key is used in byte order and has
+                //no intrinsic endianness, so write in local endian order.
+                if (BitConverter.IsLittleEndian)
```

Comment:
This looks suspicious. What does the corefx implemenation do?

Reply:
The corefx implementation uses BitCoverter.ReadInt32, and doesn't try to compensate for endianness, because in the spec the masking key isn't a number, it's just 4 bytes in order. They don't expose the masking key, though, because you can't actually write raw frames with that API.

Since we expose the masking key, I wanted both a little endian machine and a big endian machine to agree on what the actual int was that is representing the key, e.g. if it's 1234 on one side it would be 1234 on the other side.

That being said, I still got it wrong. What I actually need to do is always read/write in little endian and then swap the bytes back around before actually using the masking key if in big endian, since it gets used "raw" as a pointer into the four bytes of the int.

Reply:
OK, I just ditched this idea entirely. Trying to keep the int representation the same across platforms was just too convoluted and unclear. The code just loads and writes the bytes as-is now, same as corefx.

I did change `ToString()` on the header though to show that value as hex instead, so at the very least if you're debugging on two platforms you should see the same byte values there.